types/stylelint/index.d.ts

```diff
@@ -272,13 +273,26 @@ declare namespace stylelint {
 	type OneOrMany<S> = S | S[];
 	type Primary = number | true | OneOrMany<StringOrRegex> | Record<string, any>;
 	type Secondary = Record<string, any>;
-	type CoreRule<P extends Primary, S extends Secondary = any> = Rule<P, S>;
+
+	// see sindresorhus/type-fest#821
+	type Messages = {
+		[key in `expected${string}` | `rejected${string}`]: RuleMessage;
+	};
+
+	type AutofixMessage = { expected: (actual: string, expected: string) => string };
+
+	type CoreRule<
+		P extends Primary,
+		S extends Secondary = object,
```

Comment:
[suggestion] Minor. Is it possible to change the default for `Secondary`, like `Messages` does?

```suggestion
		S extends Secondary = Secondary,
```